consensus-types/types/payload_requests.go

```diff
@@ -60,15 +60,16 @@ func BuildNewPayloadRequestFromFork(blk *BeaconBlock) (NewPayloadRequest, error)
 	body := blk.GetBody()
 	payload := body.GetExecutionPayload()
 	parentBeaconBlockRoot := blk.GetParentBlockRoot()
-	if version.Equals(blk.GetForkVersion(), version.Deneb()) || version.Equals(blk.GetForkVersion(), version.Deneb1()) {
+	switch v := blk.GetForkVersion(); {
```

Comment:
nit: this switch case could be replaced with `if Before Electra, else`. Don't think the ForkVersionNotSupported case needs to be enforced here (generally this pattern is inconsistent throughout the code base -- so maybe a fix for a future fork version cleanup)